### ISSUE TITLE
[A11Y] [iOS only] - Inconsistent VoiceOver focus/tap behavior for items loaded after scrolling

### ISSUE THREAD

**Issue by BLKKKBVSIK** [flutter/flutter#175191]
### Steps to reproduce

1. Activate VoiceOver (only available on a real device)
2. Go over the ListView
3. Go to the last visible item on the list
4. Go to the next item (last visible item + 1)
5. Acknowledge that there is no semantic focus on the item.
6. Double tap on the screen (for making a single tap with VoiceOver)
7. Acknowledge that the tapGesture has not been recognized by Flutter. (no "tapped" in the debug console)

### Expected results

When going to the item that are not displayed in the screen before scrolling, we should have the same behavior as the previous items: the element should be focused by VoiceOver, and should be tappable.

### Actual results

When navigating with VoiceOver to an element that was not in the screen before scrolling, the element is not focused by VoiceOver and not tappable.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple)),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(backgroundColor: Theme.of(context).colorScheme.inversePrimary, title: Text(widget.title)),
      body: SafeArea(
        child: SingleChildScrollView(
          child: Column(
            children: [
              Text("Accessible Places Navigation"),
              ListView.builder(
                itemCount: 40,
                physics: const NeverScrollableScrollPhysics(),
                shrinkWrap: true,
                itemBuilder:
                    (context, index) => GestureDetector(
                      onTap: () {
                        print("Tapped");
                      },
                      child: Container(
                        padding: const EdgeInsets.all(10),
                        height: 50,
                        width: 50,
                        color: index % 2 == 0 ? Colors.blue.shade400 : Colors.blue.shade200,
                        child: Text("Item $index"),
                      ),
                    ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/29cce31c-8aba-431a-a874-7a575c7b96d5

</details>

### Logs

<details open><summary>Logs</summary>

```console
N/A No logs.
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.35.3, on macOS 15.6.1 24G90 darwin-arm64, locale en-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0-rc3)
[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.3)
[✓] VS Code (version 1.103.1)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for Елизавета’s iPhone. Ensure the
      device is unlocked and attached with a cable or associated with the same local
      area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for iPhone de Robin. Ensure the device
      is unlocked and attached with a cable or associated with the same local area
      network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@BLKKKBVSIK Thanks for the report. From the given video, I see that there's extra space which the focus moves to before moving to actual item and that could be the real issue here. https://github.com/flutter/flutter/issues/164483 is this the same one are you reporting ? I see that you have commented in it as well.

**Comment by BLKKKBVSIK** [comment]
Hey @darshankawar 

Thank you for the reply.
I don't see which extra space you are speaking about ?

#164483 is another issue that I commented on, but is a complete different behavior as this one.
I don't think they are related, as this issue only concern ListView and #164483 is related to every scrolling widget (horizontal & vertical)

**Comment by leo-mms** [comment]
@darshankawar Hello, just quickly jumping into the conversation because I'm investigating the very same issue at the moment :)

In the video there is no extra space but rather the following seems to happen: 
Before this regression was introduced in a recent Flutter version, the scroll view would ensure the next list item is fully visible and then it would render the focus border around the whole element. 
In the video, you see it vice versa: Firstly, a border is painted around the next element **but only for the part which is visible at that time**, so it's only a tiny top part of the next list item (list item 15 in the video). After this border is painted, the scroll view brings the element fully into the viewport, but then of course with a wrong border. 

This problem becomes even worse when you want to traverse to the next list item (e. g. list item 16) which is at the time of swiping not visible at all (because the scroll view aligned the previous list item 15 with the bottom of the screen). Meaning no border could be painted but I suppose something is semantically focussed internally. The scroll view continues to bring the next elements into the view port but the border could not be derived. 

I can reproduce this behavior as well.

**Comment by darshankawar** [comment]
Thanks for the analysis @leo-mms and @BLKKKBVSIK, I do see the reported behavior and occuring on iOS only.


> I don't see which extra space you are speaking about ?

I noticed this in the video you shared so wondered if you are reporting for same reason, but this seems different from it. 

```
stable : 3.35.3
master : 3.37.0-1.0.pre-66
```

**Comment by chunhtai** [comment]
Is this a regression? do you know when is the last stable version that this works?

**Comment by BLKKKBVSIK** [comment]
Hey @chunhtai.
Thank you for checking my issue

Yes, it is a regression.
It was not present in 3.29.3, but is present with 3.32.0 and onward.

Here's the behavior on 3.29.3:

https://github.com/user-attachments/assets/4aca7bc5-05c4-41bc-bbab-7f57eb7e5bf9

**Comment by chunhtai** [comment]
Thanks for the reply. I suspect this may be due to the accessibility container rect change made in march https://github.com/flutter/flutter/pull/164754 due to the rect cropping in the video

cc @LouiseHsu

Since this is iOS only issue, reassigning to iOS team.

I think this is a serious issue that probably deserve a P1 or P0 since it is a regression and will render multi item scrollview unusable in voiceover, but I will let iOS team decide

**Comment by LongCatIsLooong** [comment]
This looks like a duplicate of https://github.com/flutter/flutter/issues/173097.

**Comment by BLKKKBVSIK** [comment]
It indeed could be. Is it also a regression that is from 3.3X.X and onward ?

**Comment by LongCatIsLooong** [comment]
https://github.com/flutter/flutter/issues/173097 seems to be specific to Forms (so this is probably **not** a duplicate, but symptoms for some reason look similar). 

> I think this is a serious issue that probably deserve a P1 or P0 since it is a regression and will render multi item scrollview unusable in voiceover, but I will let iOS team decide

It looks like the widget tree used for repro has a nested scroll view where the inner scroll view uses `NeverScrollableScrollPhysics`. IIRC the two scroll views don't know how to coordinate scrolling if you're not using `NestedScrollView`. So it's probably not very well supported by design?

@BLKKKBVSIK is there a reason you're using a unscrollable scrollview instead of a Column?

**Comment by LongCatIsLooong** [comment]
Turns out this is a duplicate of https://github.com/flutter/flutter/issues/173097 since they have the same root cause.

**Comment by LongCatIsLooong** [comment]
The root cause of this issue is that the framework isn't updating the geometry of some SemanticsNodes if you, for example, wrap the child of a `SingleChildScrollView` in a single `Semantics` widget. We're working on a fix but it may take some time.

As a temporary workaround, you can move the Semantics widget out of the `SingleChildScrollView` (that is, let `Semantics` wrap the `SingleChildScrollView` instead of the other way around).
